ObjectMemoryTests.c

```diff
@@ -646,61 +646,61 @@ Test(IntegerValueWithinRange) {
-  RunTest(AllocateSmallObjectWithByteStorage);
-  RunTest(FindFirstInstanceOfClass);
-  RunTest(FindFirstLiveInstanceOfClass);
-  RunTest(FindNextInstanceOfClass);
-  RunTest(DoNotFindAnyInstanceOfClassThatWasNeverInstantiated);
-  RunTest(SwapPointers);
-  RunTest(GetIntegerValueOfObject);
-  RunTest(GetObjectForIntegerValue);
-  RunTest(IntegerValueBelowRange);
-  RunTest(IntegerValueAboveRange);
-  RunTest(IntegerValueWithinRange);
+  RunIsolatedTest(NonIntegerObjectIsNotIntegerObject);
+  RunIsolatedTest(IntegerObjectIsIntegerObject);
+  RunIsolatedTest(ObjectTableLookupFindsWordInRealWordMemory);
+  RunIsolatedTest(ObjectTableStorageSetsWordInRealWordMemory);
+  RunIsolatedTest(ObjectTableStorageRetrievesSpecificBitsInRealWordMemory);
+  RunIsolatedTest(ObjectTableStorageSetsSpecificBitsInRealWordMemory);
+  RunIsolatedTest(CountBitsInObjectStorageAreMostSignificantByteOfObjectHeader);
+  RunIsolatedTest(CountBitsStoredInMostSignificantByteOfObjectHeader);
+  RunIsolatedTest(OddBitIsAtBitEightOfObjectHeader);
+  RunIsolatedTest(OddBitIsSetAtBitEightOfObjectHeader);
+  RunIsolatedTest(PointerBitIsAtBitNineOfObjectHeader);
+  RunIsolatedTest(PointerBitIsSetAtBitNineOfObjectHeader);
+  RunIsolatedTest(FreeBitIsAtBitTenOfObjectHeader);
+  RunIsolatedTest(FreeBitIsSetAtBitTenOfObjectHeader);
+  RunIsolatedTest(SegmentBitsAtLowestNybbleOfObjectHeader);
+  RunIsolatedTest(SegmentBitsStoredInLowestNybbleOfObjectHeader);
+  RunIsolatedTest(LocationBitsAreInTheSecondWordOfObjectHeader);
+  RunIsolatedTest(LocationBitsStoredInSecondWordOfObjectHeader);
+  RunIsolatedTest(GrabbingHeapChunkOfObjectGetsWordDeeperIntoMemory);
+  RunIsolatedTest(StoringHeapChunkOfObjectWritesWordDeeperIntoMemory);
+  RunIsolatedTest(GetHeapChunkOfObjectMemoryByByte);
+  RunIsolatedTest(StoreHeapChunkOfObjectMemoryByByte);
+  RunIsolatedTest(SizeBitsOfObjectAreInFirstWordOfHeap);
+  RunIsolatedTest(StoreSizeBitsOfObjectInFirstWordOfItsHeap);
+  RunIsolatedTest(ClassBitsOfObjectAreInSecondWordOfHeap);
+  RunIsolatedTest(StoreClassBitsOfObjectInSecondWordOfItsHeap);
+  RunIsolatedTest(FetchPointerByRetrievingAppropriateWordInMemory);
+  RunIsolatedTest(StorePointerSavesInCorrectLocationAndAdjustsCounts);
+  RunIsolatedTest(FetchWordFromAppropriateLocationInMemory);
+  RunIsolatedTest(StoreWordInObjectField);
+  RunIsolatedTest(FetchLowByteOfWord);
+  RunIsolatedTest(FetchHighByteOfWord);
+  RunIsolatedTest(StoreBytesOfWord);
+  RunIsolatedTest(IncrementRefCount);
+  RunIsolatedTest(UpperBoundOfRefCount);
+  RunIsolatedTest(DecrementRefCount);
+  RunIsolatedTest(DoNotDecrementCountForMaximallyReferencedObject);
+  RunIsolatedTest(DeallocateObjectWhenFinalReferenceRemoved);
+  RunIsolatedTest(FetchClassOfRegularObject);
+  RunIsolatedTest(FetchClassOfImmediateInteger);
+  RunIsolatedTest(DiscoverWordLengthOfObject);
+  RunIsolatedTest(DiscoverByteLengthOfObjectWithEvenSize);
+  RunIsolatedTest(DiscoverByteLengthOfObjectWithOddSize);
+  RunIsolatedTest(AllocateSmallObject);
```

Comment:
This test fails, the subsequent tests all pass. My expectation is that it should pass obviously, but that also I should be able to remove the setup code from this test and it should still pass. Because the setup in `ObjectMemory_new()` should be sufficient.

Reply:
Currently that causes a different failure 🗡️